### PR TITLE
docs: document some best practices for tests

### PR DESCRIPTION
**Pull request by nhaehnle**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 00a921b253d45544303772bda3c67b760c52c8a0
<details><summary>CTS tests (Failed: 0/207274)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38905/70473 (55.2%)</li>
<li>Failed: 0/70473 (0.0%)</li>
<li>Not Supported: 31568/70473 (44.8%)</li><li>Warnings: 0/70473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70481 (50.7%)</li>
<li>Failed: 0/70481 (0.0%)</li>
<li>Not Supported: 34731/70481 (49.3%)</li><li>Warnings: 0/70481 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx9</summary><ul>

<li>Passed: 33476/66320 (50.5%)</li>
<li>Failed: 0/66320 (0.0%)</li>
<li>Not Supported: 32844/66320 (49.5%)</li><li>Warnings: 0/66320 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]
LGTM

File: docs/Contributing.md [hunk 1]
```diff
@@ -151,6 +151,19 @@ We do not want to rely on the Vulkan CTS. Running the Vulkan CTS takes a long
 time, putting a *time-to-test* burden on all developers. The Vulkan CTS tests
 are also harder to debug because more components are involved and require a GPU.
 
+LIT tests (tests that are run using the `llvm-lit` tool) should provide good
+and reliable coverage, e.g. using `ĈHECK-NEXT` lines, while still being easy
```
Comment (Flakebi) [inline docs/Contributing.md:155]: The C of `CHECK-NEXT` has a circumflex on it?

File: docs/Contributing.md [hunk 2]
```diff
@@ -151,6 +151,19 @@ We do not want to rely on the Vulkan CTS. Running the Vulkan CTS takes a long
...
+to maintain. Where reasonably possible, try to:
+
+* For passes in LGC, write an `.lgc` test that invokes `lgc` with the `-passes`
+  option to run a single pass. Use LLVM's `update_test_checks.py` with the
+  `--tool` option to automatically generate the check lines.
+* For frontend code, write a GLSL or .spvasm test (either standalone file or
```
Comment (Flakebi) [inline docs/Contributing.md:161]: .spvasm could be in backticks (as .lgc and others)

File: docs/Contributing.md (hunk 1 above)
Comment (nhaehnle) [reply docs/Contributing.md]: Dead keys... fixed.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 83558b86f80c6c1bd60f8db2bd39a68e15fc2fce
<details><summary>CTS tests (Failed: 0/207274)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 38905/70473 (55.2%)</li>
<li>Failed: 0/70473 (0.0%)</li>
<li>Not Supported: 31568/70473 (44.8%)</li><li>Warnings: 0/70473 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35750/70481 (50.7%)</li>
<li>Failed: 0/70481 (0.0%)</li>
<li>Not Supported: 34731/70481 (49.3%)</li><li>Warnings: 0/70481 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx9</summary><ul>

<li>Passed: 33476/66320 (50.5%)</li>
<li>Failed: 0/66320 (0.0%)</li>
<li>Not Supported: 32844/66320 (49.5%)</li><li>Warnings: 0/66320 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — APPROVED** [review]

**Pull request merged**